uniffi_bindgen/src/bindings/python/templates/FfiConverters.py

```diff
@@ -0,0 +1,607 @@
+        if count < 0:
+            raise InternalError("Unexpected negative sequence length")
+        items = []
+        while count > 0:
+            items.append({{ inner_type|ffi_converter_name }}.read(stream))
+            count -= 1
+        return items
+
+    @staticmethod
+    def write(builder, items):
+        builder._pack_into(4, ">i", len(items))
+        for item in items:
+            {{ inner_type|ffi_converter_name }}.write(builder, item)
+
+{% when Type::Map with (inner_type) -%}
+class {{ typ|ffi_converter_name }}:
+    # The Map<T> type for {{ inner_type.canonical_name() }}.
+
+    @classmethod
+    def lift(cls, rust_buffer):
+        with rust_buffer.consumeWithStream() as stream:
+            return cls.read(stream)
+
+    @classmethod
+    def lower(cls, v):
+        with RustBuffer.allocWithBuilder() as builder:
+            cls.write(builder, v)
+            return builder.finalize()
+
+    @staticmethod
+    def read(stream):
+        count = stream._unpack_from(4, ">i")
+        if count < 0:
+            raise InternalError("Unexpected negative map size")
+        items = {}
+        while count > 0:
+            key = FfiConverterString.read(stream)
+            items[key] = {{ inner_type|ffi_converter_name }}.read(stream)
+            count -= 1
+        return items
+
+    @staticmethod
+    def write(builder, items):
+        builder._pack_into(4, ">i", len(items))
+        for (k, v) in items.items():
+            FfiConverterString.write(builder, k)
+            {{ inner_type|ffi_converter_name }}.write(builder, v)
+
+{% when Type::Wrapped with { name, prim } -%}
+class {{ typ|ffi_converter_name }}:
+
+    @staticmethod
+    def lift(value):
+        return {{ prim|ffi_converter_name }}.lift(value)
+
+    @staticmethod
+    def lower(value):
+        return {{ prim|ffi_converter_name }}.lower(value)
+
+    @staticmethod
+    def read(stream):
+        return {{ prim|ffi_converter_name }}.read(stream)
+
+    @staticmethod
+    def write(builder, value):
+        {{ prim|ffi_converter_name }}.write(builder, value)
+
+{% when Type::External with { name, crate_name } -%}
+class {{ typ|ffi_converter_name }}:
+    @staticmethod
+    def lift(value):
+        from {{ crate_name|mod_name_py }} import {{ typ|ffi_converter_name }} as ExternalFfiConverter
+        return ExternalFfiConverter.lift(value)
+
+    @staticmethod
+    def lower(value):
+        from {{ crate_name|mod_name_py }} import {{ typ|ffi_converter_name }} as ExternalFfiConverter
+        return ExternalFfiConverter.lower(value)
+
+    @staticmethod
+    def read(stream):
+        from {{ crate_name|mod_name_py }} import {{ typ|ffi_converter_name }} as ExternalFfiConverter
+        return ExternalFfiConverter.read(stream)
+
+    @staticmethod
+    def write(builder, value):
+        from {{ crate_name|mod_name_py }} import {{ typ|ffi_converter_name }} as ExternalFfiConverter
+        ExternalFfiConverter.write(builder, value)
+
```

Comment:
The final step for the JSON work is to do the conversions on the bindings side.  My idea for that is we create the concept of an "extension crate" and define a `Type::Extension` to handle that.  The scaffolding side would be pretty much the same as the current `guid` crate.  But the extension crate could also point to a python file with an `FfiConverter` class that gets inserted here.